tests/src/Functional/EmbedFilterTest.php

```diff
@@ -0,0 +1,238 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_oembed\Functional;
+
+use Drupal\editor\Entity\Editor;
+use Drupal\file\Entity\File;
+use Drupal\filter\Entity\FilterFormat;
+use Drupal\media\Entity\Media;
+
+/**
+ * Tests the oEmbed media filter.
+ */
+class EmbedFilterTest extends EmbedTestBase {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $format = FilterFormat::create([
+      'format' => 'format_with_embed',
+      'name' => 'Format with embed',
+      'filters' => [
+        'oe_oembed_filter' => [
+          'status' => 1,
+        ],
+      ],
+    ]);
+    $format->save();
+
+    $editor_group = [
+      'name' => 'Entity Embed',
+      'items' => [
+        'media',
+        'node',
+      ],
+    ];
+
+    $editor = Editor::create([
+      'format' => 'format_with_embed',
+      'editor' => 'ckeditor',
+      'settings' => [
+        'toolbar' => [
+          'rows' => [[$editor_group]],
+        ],
+      ],
+    ]);
+    $editor->save();
+
+    $this->entityTypeManager = $this->container->get('entity_type.manager');
+    $this->createTestEntities();
+  }
+
+  /**
+   * Tests that the media oEmbed filter correctly renders the oEmbed tags.
+   */
+  public function testEmbedFilter(): void {
```

Comment:
Looks like we have missed test coverage for translated nodes.

Reply:
It's part of the current implementation. We are not yet sure of the requirements as to whether it should render in the current site language or in the node language so we keep it as is. Once we get the requiremnt from COMM to change, we adapt